File: TrainingCenter.Api/Model/Course.cs
using System.ComponentModel.DataAnnotations;
namespace TRAININGCENTER.API.Model;
public class Course
{
    
    public int id {get; set;}
    
    
     public string Title {get; set;} = string.Empty;

      public string Description {get;set;} = string.Empty;

      public int? InstructorId { get; set; } 
        
      public Instructor? instructor {get; set;}
      public int capacity {get;set;}

      public DateTime startDate {get;set;}

       public DateTime endDate {get;set;}

       public List<Enrollment> enrollmentCourse {get;set;} = new();


}

File: training-center-client/src/app/Model/Course.ts
import {instructor, InstructorResponse} from "./instructor";
import { Enrollment } from "./Enrollment";

export interface Course {
      Title: string;

      Description : string;

      instructorId? : number;

      instructor? : instructor;

      capacity : number;

      startDate : Date;

      endDate : Date;

      enrollmentCourse : Enrollment[];


}


export interface AssignInstructor
{
      InstructorId : number;
}

export interface CreateCourse {
      Title: string;

      Description : string;

      instructorId? : number;

      Capacity : number;

      StartDate : Date;

      EndDate : Date;
}

export interface UpdateCourse {
      Title: string;

      Description : string;

      InstructorId? : number;

      Capacity : number;

      StartDate : Date;

      EndDate : Date;

}
export interface CourseDetails{
        Id : number;

      Title: string;

      Description : string;

      Capacity : number;

      RegisteredStudentCount : number;

      StartDate : Date;

      EndDate : Date;
      Instructor? : InstructorResponse;
}


export interface CourseResponse{
      id : number;

      Title: string;

      Description : string;

      InstructorId? : number;

      Capacity : number;

      StartDate : Date;

      EndDate : Date;
}


